cpp/src/arrow/flight/test_definitions.cc

```diff
@@ -1469,6 +1518,17 @@ class ErrorHandlingTestServer : public FlightServerBase {
     return MakeFlightError(FlightStatusCode::Unauthorized, "Unauthorized", "extra info");
   }
 };
+
+class TransportStatusListener : public AsyncListener<FlightInfo> {
```

Comment:
Why not reuse `GetFlightInfoListener`?

Reply:
The type is slightly different, but this will change once I do the Status/StatusDetail refactor and provide the Future-returning overload.

Reply:
> The type is slightly different, but this will change once I do the Status/StatusDetail refactor and provide the Future-returning overload.

Do you intend to do this in another PR and create an issue for this?

Reply:
They're still different. I don't think the comment is relevant anymore.